hls4ml/model/optimizer/passes/infer_precision.py

```diff
@@ -1,18 +1,29 @@
 import math
-from copy import deepcopy
 
 import numpy as np
 
 from hls4ml.model.optimizer import ConfigurableOptimizerPass
 from hls4ml.model.types import FixedPrecisionType, UnspecifiedPrecisionType
 
+# TODO:  The code assumes everything is Fixed or Integer precision. Need to add checks
+
 
 class InferPrecisionTypes(ConfigurableOptimizerPass):
     def __init__(self):
         # The option, infer_no_bias, allows you to tailor for the given weights, in particular, zero bias
         self.infer_no_bias = False
+        self.count = 0
+        self.MAX_COUNT = 1000
 
     def match(self, node):
+        input_var = node.get_input_variable()
+        if input_var is not None and isinstance(input_var.type, UnspecifiedPrecisionType):
+            # need to wait for the input to update
+            # but check for infinite loops
+            self.count += 1
+            if self.count == self.MAX_COUNT:
+                raise RuntimeError("There is an infinite loop in the precision inference.")
```

Comment:
I think this is a bad idea. Detection of infinite loops should be done outside of the optimizers themselves if this ever occurs. The optimizers shouldn't hold state apart from their configuration. For example, a larger model with 1000 types to infer will break without any infinite loops taking place.

Reply:
You are right about limits potentially not working in all models. I actually think this can be removed. I think the infinite loop blocking can be made in a different way. If we want, we can check after that there is no UnspecifiedPrecisionType left, but that will produce an error further down, so I don't think it's needed.

Reply:
I'll make some changes and push.